packages/flutter/lib/src/services/restoration.dart

```diff
@@ -0,0 +1,888 @@
+  final Object debugOwner;
+
+  RestorationManager _manager;
+  RestorationBucket _parent;
+
+  /// The [RestorationId] under which the bucket is currently stored in the
+  /// parent of this bucket (or wants to be stored if it is currently
+  /// parent-less).
+  ///
+  /// This value is never null.
+  RestorationId get id => _id;
+  RestorationId _id;
+
+  Map<dynamic, dynamic> get _rawChildren => _rawData.putIfAbsent(_childrenMapKey, () => <dynamic, dynamic>{}) as Map<dynamic, dynamic>;
+  Map<dynamic, dynamic> get _rawValues => _rawData.putIfAbsent(_valuesMapKey, () => <dynamic, dynamic>{}) as Map<dynamic, dynamic>;
+
+  /// Called to signal that this bucket and all its descendants are no longer
+  /// part of the current restoration data and must not be used anymore.
+  ///
+  /// Calling this method will drop this bucket from its parent and notify all
+  /// its listeners as well as all listeners of its descendants. Once a bucket
+  /// has notified its listeners, it must not be used anymore. The bucket must
+  /// be [dispose]d and replaced with a new bucket.
+  ///
+  /// As an example, the [RestorationManager] calls this method on its root
+  /// bucket when it has been asked to restore a running application to a
+  /// different state. At that point, the data stored in the current bucket
+  /// hierarchy is invalid and will be replaced with a new hierarchy generated
+  /// from the restoration data describing the new state. To replace the current
+  /// bucket hierarchy, [decommission] is called on the root bucket to signal to
+  /// all owners of buckets in the hierarchy that their bucket has become
+  /// invalid. In response to the notification, bucket owners must [dispose]
+  /// their buckets and claim a new bucket from the newly created hierarchy. For
+  /// example, the owner of a bucket that was originally claimed from the
+  /// [RestorationManager.rootBucket] must dispose that bucket and claim a new
+  /// bucket from the new [RestorationManager.rootBucket]. Once the new bucket
+  /// is claimed, owners should restore their state according to the data stored
+  /// in the new bucket.
+  ///
+  /// A call to [decommission] must always be followed by a call to [dispose].
+  void decommission() {
+    _parent?._dropChild(this);
+    _parent = null;
+    _performDecommission();
+  }
+
+  void _performDecommission() {
+    _manager = null;
+    notifyListeners();
+    _visitChildren((RestorationBucket bucket) {
+      bucket._performDecommission();
+    });
+  }
+
+  // Get and store values.
+
+  /// Returns the value that is currently stored in the bucket under the
+  /// provided `id`.
+  ///
+  /// Returns null if nothing is stored under that id.
+  ///
+  /// See also:
+  ///
+  ///  * [put], which stores a value in the bucket.
+  ///  * [remove], which removes a value from the bucket.
+  ///  * [containsValue], which checks whether any value is stored under a given
+  ///    [RestorationId].
+  P get<P>(RestorationId id) {
```

Comment:
nit: since get is a language keyword, it might be good to avoid having it as a member name.

Reply:
https://dart.dev/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get